src/test/java/jenkins/model/TestJenkins.java

```diff
@@ -0,0 +1,37 @@
+package jenkins.model;
+
+import javax.annotation.CheckForNull;
+
+public class TestJenkins {
+    private static Jenkins.JenkinsHolder originalHolder;
+
+    /**
+     * Sets the value that {@link Jenkins#getInstance()} will return during a test.
+     * 
+     * @param mockJenkinsOrNull Null to restore the original value, otherwise
+     *                          (typically) a mock {@link Jenkins} instance.
+     */
+    public static void setJenkinsInstance(Jenkins mockJenkinsOrNull) {
+        Jenkins.JenkinsHolder current = Jenkins.HOLDER;
+        if (!(current == null || current instanceof OurJenkinsHolder)) {
+            originalHolder = current;
+        }
+        if (mockJenkinsOrNull == null) {
+            Jenkins.HOLDER = originalHolder;
+        } else {
+            Jenkins.HOLDER = new OurJenkinsHolder(mockJenkinsOrNull);
```

Comment:
Ouch, do we ABSOLUTELY need this? I failed to follow which part is problematic to stub without this.

Reply:
This is probably due to ignorance on my part, but the reason was that the `JCloudsSlave` constructor calls `AbstractCloudSlave` calls `Slave` which calls:
1. `Jenkins.getInstance().getNode(...)`
1. `Jenkins.getInstance().getLabelAtom(...)`
1. `Jenkins.getInstanceOrNull().getExtensionList(LabelFinder.class)`

I don't know how to get a "test Jenkins" set up (this aspect of Jenkins plugins has always been a mystery to me) so I mocked it.
If you know of a way it can be set up (it doesn't need to do anything other than "not throw an exception") then we could remove that test code.

TL;DR: I need the `Slave` constructor to not crash when called, which means `Jenkins.getInstance()` needs to be stubbed/mocked somehow; I'm open to suggestions on how best to do that.

Reply:
If we only need this not to blow up, running the test with `@Rule JenkinsRule` would do. most tests in the plugin does so. It runs with real Jenkins running in that JVM

Reply:
OK, I'll give that a try.

Reply:
Today I learned something :grin: 
Nasty hacky test class removed in https://github.com/jenkinsci/openstack-cloud-plugin/pull/270/commits/e0ebacc677cb89350fe1ee461973ef10cb3a1495